src/python/zquantum/core/wip/circuits/_wavefunction_operations.py

```diff
@@ -0,0 +1,46 @@
+from dataclasses import dataclass
+from numbers import Number
+
+from ._gates import _sub_symbols, Parameter
+from typing import Tuple, Sequence
+import numpy as np
+
+
+@dataclass(frozen=True)
+class MultiPhaseOperation:
+    """ TODO """
+
+    params: Tuple[Parameter, ...]
+
+    def __post_init__(self):
+        if any(
+            isinstance(param, Number) and param.imag != 0
```

Comment:
Interesting, I didn't know all Python numbers have this `.imag` property, nice!

Reply:
🐍